tethne/serialize/paper.py

```diff
@@ -0,0 +1,178 @@
+        print tethnedao.getMaxCorpusID()
+        self.corpus_id = tethnedao.getMaxCorpusID()+1
+        self.paperIdMap = {}
+        self.authorIdMap = {}
+        self.authorInstanceIdMap = {}
+
+    def serializeCorpus(self):
+        if self.corpus is not None:
+            corpus_details = [{
+                "model": "django-tethne.corpus",
+                "pk": self.corpus_id,
+                "fields": {
+                "source": self.source,
+                "date_created":strftime("%Y-%m-%d %H:%M:%S", gmtime()),
+                "length" : len(self.corpus),
+                }
+            }]
+        return corpus_details
+
+    def serializePaper(self):
+        pid = tethnedao.getMaxPaperID();
+        for paper in self.corpus:
+            pid = pid+1
+            if not hasattr(paper, 'abstract'):
+               paper.abstract = ""
+            if not hasattr(paper, 'volume'):
+                paper.volume = ""
+            if not hasattr(paper, 'title'):
+                paper.title = ""
+            if not hasattr(paper, 'date'):
+                paper.date = ""
+            if self.source == 3:
+                self.paperIdMap[paper.wosid] = pid
+            elif self.source == 2:
+                self.paperIdMap[paper.url] = pid
+
+        if self.corpus is not None:
+                papersDetails = [{
+                    "model": "django-tethne.paper",
+                    "pk": self.paperIdMap[paper.wosid],
+                    "fields": {
+                    "paper_id": paper.wosid,
+                    "corpus": self.corpus_id,
+                    "pub_date": paper.date,
+                    "volume":paper.volume,
+                    "title" : paper.title,
+                    "abstract":paper.abstract,
+                }
+            } for paper in self.corpus]
+        return papersDetails
+
+    def serializeAuthors(self):
+        author_details = []
+
+        if self.corpus is not None:
+            auid = tethnedao.getMaxAuthorID()
+            for val in self.corpus.features['authors'].index.values():
+                auid = auid+1
+                self.authorIdMap[val[1]+val[0]] = auid
+                instanceData = {
+                    "model": "django-tethne.author",
+                    "pk": auid,
+                    "fields":{
+                        "first_name": val[1],
+                        "last_name": val[0],
+                    }
+                }
+                author_details.append(instanceData)
+        return author_details
+
+
+    def serializeAuthorInstances(self):
+        author_instance_details = []
+        author_identity = []
+        auInstanceId = tethnedao.getMaxAuthorInstanceID()
+
+        if self.corpus is not None:
+            for paper in self.corpus:
+                for author in paper.authors:
+                    auInstanceId = auInstanceId+1
+                    self.authorInstanceIdMap[paper.wosid+author[0][1]+author[0][0]]=auInstanceId
```

Comment:
This is really hard to read; if you are having trouble getting it under 79 characters, create a ``key`` object on the line before, e.g.

```python
key = paper.wosid + author[0][1] + author[0][0]
self.authorInstanceIdMap[key] = auInstanceId
```